classes/page_group.php

```diff
@@ -54,6 +89,36 @@ protected function set_fuzzydurationcounts(array $data) {
         $this->raw_set('fuzzydurationcounts', json_encode($data, JSON_FORCE_OBJECT));
     }
 
+    /**
+     * Called after the page group is created in the DB.
+     */
+    public function after_create() {
+        $this->update_cache();
+    }
+
+    /**
+     * Called after the DB is updated.
+     *
+     * @param bool $result Whether or not the update was successful.
+     */
+    public function after_update($result) {
+        if ($result) {
+            $this->update_cache();
+        }
+    }
+
+    /**
+     * Updates the cache with the current record.
+     *
+     * @throws \coding_exception
```

Comment:
This method/function doesn't seem like it should have a @throws tag. If one was required, I expect to see one explicitly thrown from this method/function

Reference: https://docs.moodle.org/dev/Coding_style#.40throws